src/main/java/org/json/JSONObject.java

```diff
@@ -147,6 +147,7 @@ public String toString() {
      * The map where the JSONObject's properties are kept.
      */
     private final Map<String, Object> map;
+    public static final int RECURSION_DEPTH_LIMIT = 1000;
```

Comment:
Is this a valid value? 
Please add unit tests that exercise this value, e.g. nested JSON texts with depth 999, 1000, and 1001.

Reply:
@stleary Should I add a customization for setting the max depth based on developer preference?
For now the max depth allowed is 1000. Can I add a functionality to make such that the developer can change the max depth size to preferred value?

Reply:
`ParserConfiguration` already has a `maxNestingDepth` property that is defaulted to 512 and can be overridden by the developer. This was supposed to be used for JSONObject and JSONArray parsing in #809, but that effort appears to be stalled at the moment. To answer your question, 512 is probably a better default value. But in the long view, it might be better to go ahead and implement JSONParserConfiguration first, and just reuse the value that is already defined. Does this sound like something you would be interested in doing?

Reply:
@stleary  
Apologies for the slight delay in getting back to you
I've integrated ParserConfiguration to both JSONObject and JSONArray to enhance customization for JSON, I have implemented JSONParserConfiguration.
Let me know if this meets the required standard, or if you have any suggestion for alternative approach.